### ISSUE TITLE
Use Disco API to install different jdks distributions on demand

### ISSUE THREAD

**Issue by gochev** [jbangdev/jbang#1370]
**I would like to be able to use any Java VM and version combinations when using jbang**

I would like to be able to specify and install any Java version and jdk if wanted when using jbang. For example I might want graalvm or Azul(Zulu) or openJ9 and etc. You can use the Disco API to find the proper URL for download of every distribution.

**Describe the solution you'd like**
Use https://github.com/foojayio/discoapi so I can pass openj9 or something and get this installed if I don't have anything. Also even if I have something but want to make an example using graalVM or hotspot vs j9 I would love to be able to make the compare between them using jbang.

**Additional context**
Disco API will provide the urls for every open source jdk destribution https://github.com/foojayio/discoapi.

Also for each version like java 14 for example you can get the list of all distributions that support java 14 and so you can get that. This is super common I have to super often install different jdks versions and vendor for testing purposes and instead of using a second tool it will be great to be able to use jbang.

**Comment by gochev** [comment]
it looks it's already done just not documented and with partial possibilities

**Comment by quintesse** [comment]
Yes, you can actually install the distro of your choice by setting the `JBANG_JDK_VENDOR` environment variable to any of the values supported by Foojay's Disco API.

This is actually documented here: https://www.jbang.dev/documentation/guide/latest/usage.html but it's not something we draw a lot of attention to because it's very specific to the API. This way of doing things also only allows a single install for each major Java version. Which is quite a limitation if you want to be able to test several different distros, but we're still discussing if we want to change that or not because it would add quite a lot of complexity.

**Comment by maxandersen** [comment]
This comes up once in a while. See https://github.com/jbangdev/jbang/issues/316

I'm not against making it easier to do more via disco BUT it must not make it so the shell and power shell code gets complicated not that --java or //JAVA stops working.

**Comment by quintesse** [comment]
We're now working into this direction with work being done on #2118 (which depends on devkitman (https://github.com/jbangdev/jbang-devkitman/issues/46))  and this issue #2117
